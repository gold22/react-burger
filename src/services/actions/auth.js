export const REGISTER_USER_REQUEST = 'REGISTER_USER_REQUEST';
export const REGISTER_USER_SUCCESS = 'REGISTER_USER_SUCCESS';
export const REGISTER_USER_ERROR = 'REGISTER_USER_ERROR';

export const registerUser = (user) => async (dispatch, getState, apiClient) => {
    dispatch({
        type: REGISTER_USER_REQUEST,
    });
    try {
        const result = await apiClient.registerUser(user);
        dispatch({
            type: REGISTER_USER_SUCCESS,
            user: result.user,
            accessToken: result.accessToken,
            refreshToken: result.refreshToken,
        });
    } catch (error) {
        dispatch({
            type: REGISTER_USER_ERROR,
            message: error.message,
        });
    }
};
