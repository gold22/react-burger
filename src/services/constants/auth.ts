export const REGISTER_USER_REQUEST: 'REGISTER_USER_REQUEST' = 'REGISTER_USER_REQUEST';
export const REGISTER_USER_SUCCESS: 'REGISTER_USER_SUCCESS' = 'REGISTER_USER_SUCCESS';
export const REGISTER_USER_ERROR: 'REGISTER_USER_ERROR' = 'REGISTER_USER_ERROR';
export const LOGIN_USER_REQUEST: 'LOGIN_USER_REQUEST' = 'LOGIN_USER_REQUEST';
export const LOGIN_USER_SUCCESS: 'LOGIN_USER_SUCCESS' = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_ERROR: 'LOGIN_USER_ERROR' = 'LOGIN_USER_ERROR';
export const LOGOUT_USER_REQUEST: 'LOGOUT_USER_REQUEST' = 'LOGOUT_USER_REQUEST';
export const LOGOUT_USER_SUCCESS: 'LOGOUT_USER_SUCCESS' = 'LOGOUT_USER_SUCCESS';
export const LOGOUT_USER_ERROR: 'LOGOUT_USER_ERROR' = 'LOGOUT_USER_ERROR';
export const GET_USER_REQUEST: 'GET_USER_REQUEST' = 'GET_USER_REQUEST';
export const GET_USER_SUCCESS: 'GET_USER_SUCCESS' = 'GET_USER_SUCCESS';
export const GET_USER_ERROR: 'GET_USER_ERROR' = 'GET_USER_ERROR';
export const UPDATE_USER_REQUEST: 'UPDATE_USER_REQUEST' = 'UPDATE_USER_REQUEST';
export const UPDATE_USER_SUCCESS: 'UPDATE_USER_SUCCESS' = 'UPDATE_USER_SUCCESS';
export const UPDATE_USER_ERROR: 'UPDATE_USER_ERROR' = 'UPDATE_USER_ERROR';
