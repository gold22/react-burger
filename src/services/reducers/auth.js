import {
    REGISTER_USER_REQUEST,
    REGISTER_USER_SUCCESS,
    REGISTER_USER_ERROR,
    LOGIN_USER_REQUEST,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
    LOGOUT_USER_REQUEST,
    LOGOUT_USER_SUCCESS,
    LOGOUT_USER_ERROR,
    GET_USER_REQUEST,
    GET_USER_SUCCESS,
    GET_USER_ERROR,
    UPDATE_USER_REQUEST,
    UPDATE_USER_SUCCESS,
    UPDATE_USER_ERROR,
    SET_REDIRECT_TO,
} from '../actions/auth';

const unauthState = {
    user: null,
    isRegistering: false,
    registrationError: null,
    isLoggingIn: false,
    loginError: null,
    isLoggingOut: false,
    logoutError: null,
    isSendingEmail: false,
    sendingEmailError: null,
    isResettingPassword: false,
    resetPasswordError: null,
    isLoading: true,
    loadError: null,
    isUpdating: true,
    updateError: null,
};

const initialState = {
    redirectTo: '/',
    ...unauthState,
};

// eslint-disable-next-line import/prefer-default-export
export const authReducer = (state = initialState, action) => {
    switch (action.type) {
    case REGISTER_USER_REQUEST: {
        return {
            ...state,
            isRegistering: true,
        };
    }
    case REGISTER_USER_SUCCESS: {
        return {
            ...state,
            ...unauthState,
            user: action.user,
        };
    }
    case REGISTER_USER_ERROR: {
        return {
            ...state,
            ...unauthState,
            registrationError: action.message,
        };
    }
    case LOGIN_USER_REQUEST: {
        return {
            ...state,
            isLoggingIn: true,
        };
    }
    case LOGIN_USER_SUCCESS: {
        return {
            ...state,
            ...unauthState,
            user: action.user,
        };
    }
    case LOGIN_USER_ERROR: {
        return {
            ...state,
            ...unauthState,
            loginError: action.message,
        };
    }
    case LOGOUT_USER_REQUEST: {
        return {
            ...state,
            isLoggingOut: true,
            logoutError: null,
        };
    }
    case LOGOUT_USER_SUCCESS: {
        return {
            ...initialState,
        };
    }
    case LOGOUT_USER_ERROR: {
        return {
            ...state,
            isLoggingOut: false,
            logoutError: action.message,
        };
    }
    case GET_USER_REQUEST: {
        return {
            ...state,
            isLoading: true,
        };
    }
    case GET_USER_SUCCESS: {
        return {
            ...state,
            user: action.user,
            isLoading: false,
            loadError: null,
        };
    }
    case GET_USER_ERROR: {
        return {
            ...state,
            isLoading: false,
            loadError: action.message,
        };
    }
    case UPDATE_USER_REQUEST: {
        return {
            ...state,
            isUpdating: true,
        };
    }
    case UPDATE_USER_SUCCESS: {
        return {
            ...state,
            user: action.user,
            isUpdating: false,
            updateError: null,
        };
    }
    case UPDATE_USER_ERROR: {
        return {
            ...state,
            isUpdating: false,
            updateError: action.message,
        };
    }
    case SET_REDIRECT_TO: {
        return {
            ...state,
            redirectTo: action.redirectTo,
        };
    }
    default: {
        return state;
    }
    }
};
