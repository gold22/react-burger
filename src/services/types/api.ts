export type TApiConfig = {
    url: string;
    wsUrl: string;
};

export type TApiIngredient = {
    _id: string;
    name: string;
    type: string;
    proteins: number;
    fat: number;
    carbohydrates: number;
    calories: number;
    price: number;
    image: string;
    image_large: string;
    image_mobile: string;
};

export type TApiIngredients = Array<TApiIngredient>;

export type TApiOrderInfo = {
    name: string;
    order: { number: string };
};

export enum TApiOrderStatus {
    Pending = 'pending',
    Created = 'created',
    Done = 'done',
}

export type TApiOrder = {
    number: number;
    name: string;
    status: TApiOrderStatus;
    createdAt: string;
    updatedAt: string;
    ingredients: Array<string>;
};

export type TApiOrders = {
    orders: Array<TApiOrder>;
    total: number;
    totalToday: number;
};

export type TApiUser = {
    name: string;
    email: string;
    password: string;
};

export type TApiUserInfo = {
    name: string;
    email: string;
};

export type TApiUserCredentials = {
    email: string;
    password: string;
};

export type TApiUserResetPasswordInfo = {
    password: string;
    token: string;
};
