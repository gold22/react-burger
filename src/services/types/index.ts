import { Location } from 'history';

export type TIngredient = {
    uuid: string;
    id: string;
    name: string;
    type: string;
    proteins: number;
    fat: number;
    carbohydrates: number;
    calories: number;
    price: number;
    image: string;
    imageLarge: string;
    imageMobile: string;
};

export type TIngredients = Array<TIngredient>;

export type TIngredientDragItem = {
    ingredient?: TIngredient;
    index?: number;
};

export type TLocationState = {
    background?: Location;
    from?: Location;
};

export enum IngredientType {
    Bun = 'bun',
    Sauce = 'sauce',
    Main = 'main',
}
