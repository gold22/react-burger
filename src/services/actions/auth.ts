import {
    REGISTER_USER_REQUEST,
    REGISTER_USER_SUCCESS,
    REGISTER_USER_ERROR,
    LOGIN_USER_REQUEST,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
    LOGOUT_USER_REQUEST,
    LOGOUT_USER_SUCCESS,
    LOGOUT_USER_ERROR,
    GET_USER_REQUEST,
    GET_USER_SUCCESS,
    GET_USER_ERROR,
    UPDATE_USER_REQUEST,
    UPDATE_USER_SUCCESS,
    UPDATE_USER_ERROR,
} from '../constants/auth';
import { TApiUser, TApiUserCredentials } from '../types/api';
import { TDispatch, TGetState, TThunk } from '../types/store';
import ApiClient from '../api-client';
import { getErrorMessage } from '../../utils/error';

export const registerUser: TThunk = (user: TApiUser) => async (
    dispatch: TDispatch,
    getState: TGetState,
    apiClient: ApiClient,
) => {
    dispatch({
        type: REGISTER_USER_REQUEST,
    });
    try {
        const result = await apiClient.registerUser(user);
        dispatch({
            type: REGISTER_USER_SUCCESS,
            user: result,
        });
    } catch (error) {
        dispatch({
            type: REGISTER_USER_ERROR,
            message: getErrorMessage(error),
        });
    }
};

export const logInUser: TThunk = (user: TApiUserCredentials) => async (
    dispatch: TDispatch,
    getState: TGetState,
    apiClient: ApiClient,
) => {
    dispatch({
        type: LOGIN_USER_REQUEST,
    });
    try {
        const result = await apiClient.logInUser(user);
        dispatch({
            type: LOGIN_USER_SUCCESS,
            user: result,
        });
    } catch (error) {
        dispatch({
            type: LOGIN_USER_ERROR,
            message: getErrorMessage(error),
        });
    }
};

export const logOutUser: TThunk = () => async (
    dispatch: TDispatch,
    getState: TGetState,
    apiClient: ApiClient,
) => {
    dispatch({
        type: LOGOUT_USER_REQUEST,
    });
    try {
        await apiClient.logOutUser();
        dispatch({
            type: LOGOUT_USER_SUCCESS,
        });
    } catch (error) {
        dispatch({
            type: LOGOUT_USER_ERROR,
            message: getErrorMessage(error),
        });
    }
};

export const getUser = () => async (
    dispatch: TDispatch,
    getState: TGetState,
    apiClient: ApiClient,
) => {
    dispatch({
        type: GET_USER_REQUEST,
    });
    try {
        const user = await apiClient.getUser();
        dispatch({
            type: GET_USER_SUCCESS,
            user,
        });
    } catch (error) {
        dispatch({
            type: GET_USER_ERROR,
            message: getErrorMessage(error),
        });
    }
};

export const updateUser = (user: TApiUser) => async (
    dispatch: TDispatch,
    getState: TGetState,
    apiClient: ApiClient,
) => {
    dispatch({
        type: UPDATE_USER_REQUEST,
    });
    try {
        const result = await apiClient.updateUser(user);
        dispatch({
            type: UPDATE_USER_SUCCESS,
            user: result,
        });
    } catch (error) {
        dispatch({
            type: UPDATE_USER_ERROR,
            message: getErrorMessage(error),
        });
    }
};
