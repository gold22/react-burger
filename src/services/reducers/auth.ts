import {
    REGISTER_USER_REQUEST,
    REGISTER_USER_SUCCESS,
    REGISTER_USER_ERROR,
    LOGIN_USER_REQUEST,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
    LOGOUT_USER_REQUEST,
    LOGOUT_USER_SUCCESS,
    LOGOUT_USER_ERROR,
    GET_USER_REQUEST,
    GET_USER_SUCCESS,
    GET_USER_ERROR,
    UPDATE_USER_REQUEST,
    UPDATE_USER_SUCCESS,
    UPDATE_USER_ERROR,
} from '../constants/auth';
import { TApiUserInfo } from '../types/api';
import { TAuthActions } from '../types/actions/auth';

export type TAuthState = {
    user: TApiUserInfo | null;
    isRegistering: boolean;
    registrationError: string | null;
    isLoggingIn: boolean;
    loginError: string | null;
    isLoggingOut: boolean;
    logoutError: string | null;
    isLoading: boolean;
    loadError: string | null;
    isUpdating: boolean;
    updateError: string | null;
};

const initialState: TAuthState = {
    user: null,
    isRegistering: false,
    registrationError: null,
    isLoggingIn: false,
    loginError: null,
    isLoggingOut: false,
    logoutError: null,
    isLoading: true,
    loadError: null,
    isUpdating: true,
    updateError: null,
};

// eslint-disable-next-line @typescript-eslint/default-param-last
export const authReducer = (state: TAuthState = initialState, action: TAuthActions): TAuthState => {
    switch (action.type) {
    case REGISTER_USER_REQUEST: {
        return {
            ...state,
            isRegistering: true,
        };
    }
    case REGISTER_USER_SUCCESS: {
        return {
            ...initialState,
            user: action.user,
        };
    }
    case REGISTER_USER_ERROR: {
        return {
            ...initialState,
            registrationError: action.message,
        };
    }
    case LOGIN_USER_REQUEST: {
        return {
            ...state,
            isLoggingIn: true,
        };
    }
    case LOGIN_USER_SUCCESS: {
        return {
            ...initialState,
            user: action.user,
        };
    }
    case LOGIN_USER_ERROR: {
        return {
            ...initialState,
            loginError: action.message,
        };
    }
    case LOGOUT_USER_REQUEST: {
        return {
            ...state,
            isLoggingOut: true,
            logoutError: null,
        };
    }
    case LOGOUT_USER_SUCCESS: {
        return {
            ...initialState,
        };
    }
    case LOGOUT_USER_ERROR: {
        return {
            ...state,
            isLoggingOut: false,
            logoutError: action.message,
        };
    }
    case GET_USER_REQUEST: {
        return {
            ...state,
            isLoading: true,
        };
    }
    case GET_USER_SUCCESS: {
        return {
            ...state,
            user: action.user,
            isLoading: false,
            loadError: null,
        };
    }
    case GET_USER_ERROR: {
        return {
            ...state,
            isLoading: false,
            loadError: action.message,
        };
    }
    case UPDATE_USER_REQUEST: {
        return {
            ...state,
            isUpdating: true,
        };
    }
    case UPDATE_USER_SUCCESS: {
        return {
            ...state,
            user: action.user,
            isUpdating: false,
            updateError: null,
        };
    }
    case UPDATE_USER_ERROR: {
        return {
            ...state,
            isUpdating: false,
            updateError: action.message,
        };
    }
    default: {
        return state;
    }
    }
};
