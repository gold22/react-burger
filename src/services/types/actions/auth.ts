import {
    GET_USER_ERROR,
    GET_USER_REQUEST,
    GET_USER_SUCCESS,
    LOGIN_USER_ERROR,
    LOGIN_USER_REQUEST,
    LOGIN_USER_SUCCESS,
    LOGOUT_USER_ERROR,
    LOGOUT_USER_REQUEST,
    LOGOUT_USER_SUCCESS,
    REGISTER_USER_ERROR,
    REGISTER_USER_REQUEST,
    REGISTER_USER_SUCCESS,
    UPDATE_USER_ERROR,
    UPDATE_USER_REQUEST,
    UPDATE_USER_SUCCESS,
} from '../../constants/auth';
import { TApiUserInfo } from '../api';

export interface IRegisterUserAction {
    readonly type: typeof REGISTER_USER_REQUEST;
}

export interface IRegisterUserSuccessAction {
    readonly type: typeof REGISTER_USER_SUCCESS;
    readonly user: TApiUserInfo;
}

export interface IRegisterUserFailedAction {
    readonly type: typeof REGISTER_USER_ERROR;
    readonly message: string;
}

export interface ILogInUserAction {
    readonly type: typeof LOGIN_USER_REQUEST;
}

export interface ILogInUserSuccessAction {
    readonly type: typeof LOGIN_USER_SUCCESS;
    readonly user: TApiUserInfo;
}

export interface ILogInUserFailedAction {
    readonly type: typeof LOGIN_USER_ERROR;
    readonly message: string;
}

export interface ILogOutUserAction {
    readonly type: typeof LOGOUT_USER_REQUEST;
}

export interface ILogOutUserSuccessAction {
    readonly type: typeof LOGOUT_USER_SUCCESS;
}

export interface ILogOutUserFailedAction {
    readonly type: typeof LOGOUT_USER_ERROR;
    readonly message: string;
}

export interface IGetUserAction {
    readonly type: typeof GET_USER_REQUEST;
}

export interface IGetUserSuccessAction {
    readonly type: typeof GET_USER_SUCCESS;
    readonly user: TApiUserInfo;
}

export interface IGetUserFailedAction {
    readonly type: typeof GET_USER_ERROR;
    readonly message: string;
}

export interface IUpdateUserAction {
    readonly type: typeof UPDATE_USER_REQUEST;
}

export interface IUpdateUserSuccessAction {
    readonly type: typeof UPDATE_USER_SUCCESS;
    readonly user: TApiUserInfo;
}

export interface IUpdateUserFailedAction {
    readonly type: typeof UPDATE_USER_ERROR;
    readonly message: string;
}

export type TAuthActions =
    | IRegisterUserAction
    | IRegisterUserSuccessAction
    | IRegisterUserFailedAction
    | ILogInUserAction
    | ILogInUserSuccessAction
    | ILogInUserFailedAction
    | ILogOutUserAction
    | ILogOutUserSuccessAction
    | ILogOutUserFailedAction
    | IGetUserAction
    | IGetUserSuccessAction
    | IGetUserFailedAction
    | IUpdateUserAction
    | IUpdateUserSuccessAction
    | IUpdateUserFailedAction;
