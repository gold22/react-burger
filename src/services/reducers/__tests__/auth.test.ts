import { authReducer, TAuthState } from '../auth';
import { TApiUserInfo } from '../../types/api';
import {
    REGISTER_USER_REQUEST,
    REGISTER_USER_SUCCESS,
    REGISTER_USER_ERROR,
    LOGIN_USER_REQUEST,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
    LOGOUT_USER_REQUEST,
    LOGOUT_USER_SUCCESS,
    LOGOUT_USER_ERROR,
    GET_USER_REQUEST,
    GET_USER_SUCCESS,
    GET_USER_ERROR,
    UPDATE_USER_REQUEST,
    UPDATE_USER_SUCCESS,
    UPDATE_USER_ERROR,
} from '../../constants/auth';
import { userInfo } from '../../../__fixtures__/user';

const initialState: TAuthState = authReducer(undefined, {} as any);

describe('auth reducer', () => {
    it('should return the initial state', () => {
        expect(initialState).toMatchSnapshot();
    });

    it('should handle REGISTER_USER_REQUEST', () => {
        expect(authReducer(undefined, { type: REGISTER_USER_REQUEST }))
            .toMatchSnapshot();
    });

    it('should handle REGISTER_USER_SUCCESS', () => {
        const state = {
            ...initialState,
            isRegistering: true,
        };
        expect(authReducer(state, { type: REGISTER_USER_SUCCESS, user: userInfo }))
            .toMatchSnapshot();
    });

    it('should handle REGISTER_USER_ERROR', () => {
        const state = {
            ...initialState,
            isRegistering: true,
        };
        expect(authReducer(state, { type: REGISTER_USER_ERROR, message: 'Registration error' }))
            .toMatchSnapshot();
    });

    it('should handle LOGIN_USER_REQUEST', () => {
        expect(authReducer(undefined, { type: LOGIN_USER_REQUEST }))
            .toMatchSnapshot();
    });

    it('should handle LOGIN_USER_SUCCESS', () => {
        const state = {
            ...initialState,
            isLoggingIn: true,
        };
        expect(authReducer(state, { type: LOGIN_USER_SUCCESS, user: userInfo }))
            .toMatchSnapshot();
    });

    it('should handle LOGIN_USER_ERROR', () => {
        const state = {
            ...initialState,
            isLoggingIn: true,
        };
        expect(authReducer(state, { type: LOGIN_USER_ERROR, message: 'Login error' }))
            .toMatchSnapshot();
    });

    it('should handle LOGOUT_USER_REQUEST', () => {
        const state = {
            ...initialState,
            user: userInfo,
        };
        expect(authReducer(state, { type: LOGOUT_USER_REQUEST }))
            .toMatchSnapshot();
    });

    it('should handle LOGOUT_USER_SUCCESS', () => {
        const state = {
            ...initialState,
            user: userInfo,
            isLoggingOut: true,
        };
        expect(authReducer(state, { type: LOGOUT_USER_SUCCESS }))
            .toMatchSnapshot();
    });

    it('should handle LOGOUT_USER_ERROR', () => {
        const state = {
            ...initialState,
            user: userInfo,
            isLoggingOut: true,
        };
        expect(authReducer(state, { type: LOGOUT_USER_ERROR, message: 'Logout error' }))
            .toMatchSnapshot();
    });

    it('should handle GET_USER_REQUEST', () => {
        expect(authReducer(undefined, { type: GET_USER_REQUEST }))
            .toMatchSnapshot();
    });

    it('should handle GET_USER_SUCCESS', () => {
        const state = {
            ...initialState,
            isLoading: true,
        };
        expect(authReducer(state, { type: GET_USER_SUCCESS, user: userInfo }))
            .toMatchSnapshot();
    });

    it('should handle GET_USER_ERROR', () => {
        const state = {
            ...initialState,
            isLoading: true,
        };
        expect(authReducer(state, { type: GET_USER_ERROR, message: 'Get user error' }))
            .toMatchSnapshot();
    });

    it('should handle UPDATE_USER_REQUEST', () => {
        const state = {
            ...initialState,
            user: userInfo,
        };
        expect(authReducer(state, { type: UPDATE_USER_REQUEST }))
            .toMatchSnapshot();
    });

    it('should handle UPDATE_USER_SUCCESS', () => {
        const state = {
            ...initialState,
            isUpdating: true,
        };
        const updatedUserInfo: TApiUserInfo = {
            name: 'Updated',
            email: 'updated@example.tld',
        };
        expect(authReducer(state, { type: UPDATE_USER_SUCCESS, user: updatedUserInfo }))
            .toMatchSnapshot();
    });

    it('should handle UPDATE_USER_ERROR', () => {
        const state = {
            ...initialState,
            isUpdating: true,
        };
        expect(authReducer(state, { type: UPDATE_USER_ERROR, message: 'Update user error' }))
            .toMatchSnapshot();
    });
});
