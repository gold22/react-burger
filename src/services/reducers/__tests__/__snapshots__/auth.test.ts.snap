// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer should handle GET_USER_ERROR 1`] = `
Object {
  "isLoading": false,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": "Get user error",
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle GET_USER_REQUEST 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle GET_USER_SUCCESS 1`] = `
Object {
  "isLoading": false,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle LOGIN_USER_ERROR 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": "Login error",
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle LOGIN_USER_REQUEST 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": true,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle LOGIN_USER_SUCCESS 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle LOGOUT_USER_ERROR 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": "Logout error",
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle LOGOUT_USER_REQUEST 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": true,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle LOGOUT_USER_SUCCESS 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle REGISTER_USER_ERROR 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": "Registration error",
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle REGISTER_USER_REQUEST 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": true,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;

exports[`auth reducer should handle REGISTER_USER_SUCCESS 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle UPDATE_USER_ERROR 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": false,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": "Update user error",
  "user": null,
}
`;

exports[`auth reducer should handle UPDATE_USER_REQUEST 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "test@example.tld",
    "name": "Test",
  },
}
`;

exports[`auth reducer should handle UPDATE_USER_SUCCESS 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": false,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": Object {
    "email": "updated@example.tld",
    "name": "Updated",
  },
}
`;

exports[`auth reducer should return the initial state 1`] = `
Object {
  "isLoading": true,
  "isLoggingIn": false,
  "isLoggingOut": false,
  "isRegistering": false,
  "isResettingPassword": false,
  "isSendingEmail": false,
  "isUpdating": true,
  "loadError": null,
  "loginError": null,
  "logoutError": null,
  "registrationError": null,
  "resetPasswordError": null,
  "sendingEmailError": null,
  "updateError": null,
  "user": null,
}
`;
